### ISSUE TITLE
Payment captured but invoice not generated

### ISSUE THREAD

**Issue by cowell-manhnt7190** [Midtrans/Midtrans-Magento2#32]
**Preconditions and environment**

- Magento 2.4.3 and Midtrans 2.6.5

**Steps to reproduce**

- Open web, add a product to Cart
- Go to checkout, payment page
- Input incorrect data: 4411 1111 1111 1118 , 10/25, CVV: 111
- Input correct data again: 4411 1111 1111 1118 , 10/25, CVV: 123

**Expected result**

- Order status change to processing and invoice generated.

**Actual result**

- Order status is processing and invoice is not generated.

Hint maybe:
When Input data is incorrect, order received Deny notification from midtrands, and order status change to Payment Review. After customer input correct data again, order received Capture notification from midtrands. However, the invoice is not generated because the order did not pass canInvoice function because the current status is Payment Review.

**Comment by Xaxxis** [comment]
Hi @cowell-manhnt7190 thank you for reaching out to us. This is an acknowledgeable bug/issue. Currently still on the development to fix and hopefully can release this week. 

Anyway, for a temporary workaround. May you can change these line code 
- https://github.com/Midtrans/Midtrans-Magento2/blob/master/Controller/Payment/Notification.php#L89-L93
- https://github.com/Midtrans/Midtrans-Magento2/blob/master/Controller/Payment/Notification.php#L98-L102

 from this
```
     $order_note = $note_prefix . 'Payment Completed - ' . $payment_type;
     if ($order->canInvoice()) {
         $this->paymentOrderRepository->generateInvoice($order);
     }
    $this->paymentOrderRepository->setOrderStateAndStatus($order, Order::STATE_PROCESSING, $order_note);
```
to this
```
     $order_note = $note_prefix . 'Payment Completed - ' . $payment_type;
     $this->paymentOrderRepository->setOrderStateAndStatus($order, Order::STATE_PROCESSING, $order_note);
     if ($order->canInvoice()) {
         $this->paymentOrderRepository->generateInvoice($order);
     }
```

Thank you

**Comment by cowell-manhnt7190** [comment]
Hi @Xaxxis Thank you for your information. Hope there will be an update soon.

**Comment by Xaxxis** [comment]
The fix is already on Pull request https://github.com/Midtrans/Midtrans-Magento2/pull/33 and is still in progress testing. We will let you know once release, thanks

**Comment by Xaxxis** [comment]
Hi @cowell-manhnt7190 the fix is already released on v2.7.0. Please kindly update Midtrans plugins using v2.7.0
If you have any general questions may you can reach out to us via Midtrans contact us page [here](https://midtrans.com/contact-us)

Thank you

**Comment by cowell-manhnt7190** [comment]
Hi @Xaxxis Thank you so much for your support!
